woocommerce/woocommerce
# Fix #32542

Closes #32542. See notes in that issue for more context.

### Expected behavior

* Upon placing an order #xxx for a variation with a taxonomy product attribute, the order item meta value stored in the database is the taxonomy product attribute **term slug**.
* Upon using the "Order again" button for order #xxx and placing the order #yyy, the order item meta value stored in the database is the taxonomy product attribute **term slug**.

### Actual behavior

* Upon placing an order #xxx for a variation with a taxonomy product attribute, the order item meta value stored in the database is the taxonomy product attribute **term slug**.
* Upon using the "Order again" button for order #xxx and placing the order #yyy, the order item meta value stored in the database is the taxonomy product attribute **term name**.

### Steps to reproduce

For:
* a product attribute "Test Attribute" with `slug:value` tuples:
    *  `test1:TEST1`
    *  `test2:TEST2`
* a variable product "Test Product"
* a product variation "Test Variation" with attribute `test1:TEST1`
* a product variation "Test Variation" with attribute `test2:TEST2`

Follow the steps below from the `/shop` page:
* Click "Select options" for product "Test Product"
* Select "Test Attribute" > "TEST1"
* Click "Add to cart"
* Click "view cart" > "Proceed to checkout"
* Place the order #xxx with "Cash on delivery" payment method


Follow the steps below from the `/wp-admin/edit.php?post_type=shop_order` page:
* Ensure order #xxx  is "Processing"
* Edit the order #xxx status to "Completed"

Follow the steps below from the `/my-account/orders` page:
* Click "View" for order #xxx
* Click "Order again" button
* Click "view cart" > "Proceed to checkout"
* Place the order #yyy with "Cash on delivery" payment method

Follow the steps below from the `/wp-admin/edit.php?post_type=shop_order` page:
* Ensure order #yyy  is "Processing"
* Edit the order #yyy status to "Completed"

With phpMyAdmin or an RDBMS:
* Connect to the WordPress database
* Execute the following query:

```sql
SELECT * FROM `wp_woocommerce_order_itemmeta` WHERE `meta_value` LIKE '%test%'
```

* See the following result (ids have been edited out in the sample below):

```
meta_id | order_item_id | meta_key | meta_value
-------- | -------------- | ---------- | ------------
1 | xxx | 'pa_test-attribute' | 'test1'
2 | yyy | 'pa_test-attribute' | 'TEST1'
```

### Other information:




### FOR PR REVIEWER ONLY:
